app/jobs/result_monitor_loader.rb

```diff
@@ -0,0 +1,19 @@
+# Load a result from S3 into the db
+require 'logger'
+class ResultMonitorLoader
+  @queue = :q03_pipeline_run
+  @logger = Logger.new(STDOUT)
+
+  def self.perform(pipeline_run_id, load_db_command_func)
+    @logger.info("#{load_db_command_func} for pipeline run #{pipeline_run_id}")
+    pr = PipelineRun.find(pipeline_run_id)
+    begin
+      pr.send(load_db_command_func) unless pr.completed?
```

Comment:
This reads like a bug to me.    Seems you want to skip doing this work if it's been done before.   If `pr.completed?` is the signal for that, then you should skip everything conditional on that signal, not just the `pr.send(load_db_command_func)` but also the `pr.update_result_status`?

Reply:
Also I can't seem to find the enque for this anywhere.

Reply:
It's in the file whose diff was hidden because it is too large ;)

Reply:
agree on the conditional skipping